src/v/storage/kvstore.h

```diff
@@ -137,6 +139,7 @@ class kvstore {
     ss::abort_source _as;
     simple_snapshot_manager _snap;
     bool _started{false};
+    std::vector<ss::sstring> _startup_garbage;
```

Comment:
depending on the expected size, it might be better to use utils/fragmented_vector.h

Reply:
Ya, I don't know the expected size here. Hopefully someone else can comment.

Reply:
honestly it is probably fine because:
1. this happens on startup and memory is not fragmented
2. we clear it after use so memory **should** be freed, but I believe `clear` doesn't actually free the underlying memory in the vector, you either need to swap it out for a default constructed version or call `shrink_to_fit()`.

Might be safer to just use `ss::chunked_fifo` as a general rule of thumb if you don't need random access.